### PR TITLE
Fixes #2225. MenuBar leaves trails when mounted in a Window.

### LINKED ISSUE #2225: MenuBar leaves trails when mounted in a `Window`
**Describe the bug**
When mounting a `MenuBar` in a `Window` you get a trail left behind where underlying window does not update itself to clear the old menu.

This seems to be the behaviour in both `1.8.2` nuget package and on `develop`.

Resizing the window clears the old 'trailing' state:

menubar-trails

**To Reproduce**
```csharp
Application.Init ();
MenuBar menubar;
MenuBarItem fileF9;
MenuItem runnewTask;
MenuItem exit;
MenuBarItem view;
MenuItem refreshNow;
MenuBarItem updateSpeed;
MenuItem high;
MenuItem normal;
MenuItem slow;
MenuItem paused;
var win = new Window ();
menubar = new Terminal.Gui.MenuBar ();
win.Width = Dim.Fill (0);
win.Height = Dim.Fill (0);
win.X = 0;
win.Y = 0;
win.Modal = false;
win.Text = "";
win.Border.BorderStyle = Terminal.Gui.BorderStyle.Single;
win.Border.BorderBrush = Terminal.Gui.Color.Blue;
win.Border.Effect3D = false;
win.Border.Effect3DBrush = null;
win.Border.DrawMarginFrame = true;
win.TextAlignment = Terminal.Gui.TextAlignment.Left;
win.Title = "Task Manager";
menubar.Width = Dim.Fill (0);
menubar.Height = 1;
menubar.X = 0;
menubar.Y = 0;
menubar.Data = "menubar";
menubar.Text = "";
menubar.TextAlignment = Terminal.Gui.TextAlignment.Left;
fileF9 = new Terminal.Gui.MenuBarItem ();
fileF9.Title = "_File (F9)";
runnewTask = new Terminal.Gui.MenuItem ();
runnewTask.Title = "Run new Task...";
runnewTask.Data = "RunnewTask";
runnewTask.Shortcut = ((Terminal.Gui.Key)(0u));
exit = new Terminal.Gui.MenuItem ();
exit.Title = "Exit";
exit.Data = "Exit";
exit.Shortcut = ((Terminal.Gui.Key)(0u));
fileF9.Children = new Terminal.Gui.MenuItem [] {
		runnewTask,
		exit};
view = new Terminal.Gui.MenuBarItem ();
view.Title = "View";
refreshNow = new Terminal.Gui.MenuItem ();
refreshNow.Title = "Refresh Now";
refreshNow.Data = "RefreshNow";
refreshNow.Shortcut = ((Terminal.Gui.Key)(0u));
updateSpeed = new Terminal.Gui.MenuBarItem ();
updateSpeed.Title = "Update Speed";
high = new Terminal.Gui.MenuItem ();
high.Title = "High";
high.Data = "High";
high.Shortcut = ((Terminal.Gui.Key)(0u));
normal = new Terminal.Gui.MenuItem ();
normal.Title = "Normal";
normal.Data = "Normal";
normal.Shortcut = ((Terminal.Gui.Key)(0u));
slow = new Terminal.Gui.MenuItem ();
slow.Title = "Slow";
slow.Data = "Slow";
slow.Shortcut = ((Terminal.Gui.Key)(0u));
paused = new Terminal.Gui.MenuItem ();
paused.Title = "Paused";
paused.Data = "Paused";
paused.Shortcut = ((Terminal.Gui.Key)(0u));
updateSpeed.Children = new Terminal.Gui.MenuItem [] {
		high,
		normal,
		slow,
		paused};
view.Children = new Terminal.Gui.MenuItem [] {
		refreshNow,
		updateSpeed};
menubar.Menus = new Terminal.Gui.MenuBarItem [] {
		fileF9,
		view};
win.Add (menubar);
Application.Run (win);
```

### LINKED ISSUE #2238: MenuBar vanishes if View underneath repaints
**Describe the bug**
Not sure if this is related to #2225 but if you have a `MenuBar` and beneath it a `View` that is updating itself periodically (e.g. a `TableView`) then the repaint of the `TableView` occurs over the top of any open menu.

menu-no-repaint

**To Reproduce**
1. Add a MenuBar
2. Add a View that fills the space
3. Add a timeout that calls `SetNeedsDisplay` on the big view every 1s
4. Open the menu

**Expected behavior**
Below view not to render over the top of an open menu

 - Version: Nuget Package 1.8.2

### PR DESCRIPTION
**Pull request by BDisp**
Fixes #2225 - The Window and his sub-views weren't marked with `NeedDisplay`.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

Also Fixes #2238.

### CODE REVIEW

**Comment by tig** [comment]
@BDisp all the PR merging has caused a conflict on this one. Can you please fix?

File: Terminal.Gui/Windows/FileDialog.cs [hunk 1]
```diff
@@ -116,11 +116,17 @@ protected override void Dispose (bool disposing)
 
 		void Watcher_Error (object sender, ErrorEventArgs e)
 		{
+			if (Application.MainLoop == null)
```
Comment (tig) [inline Terminal.Gui/Windows/FileDialog.cs:119]: 1) This fixes an issue not related to this PR. Please remove from this PR and do another one.
2) If statement needs braces.

**Review by tig — CHANGES_REQUESTED** [review]
See my comments

File: Terminal.Gui/Core/Window.cs [hunk 2]
```diff
@@ -280,15 +280,16 @@ public override void Redraw (Rect bounds)
 			//var borderLength = Border.DrawMarginFrame ? 1 : 0;
 
 			// FIXED: Why do we draw the frame twice? This call is here to clear the content area, I think. Why not just clear that area?
```
[the remark this answers is not in the document]
Comment (BDisp) [reply Terminal.Gui/Core/Window.cs]: Already removed the comments.

File: Terminal.Gui/Windows/FileDialog.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Windows/FileDialog.cs]: @tig this change is already belong to the develop branch and if it's really need to change also would be need to change this:

imagem

Do you really think it worth it?

**Comment by BDisp** [comment]
> @BDisp all the PR merging has caused a conflict on this one. Can you please fix?

@tig I already resolved the conflicts.

**Review by tig — APPROVED** [review]

**Pull request merged**